### PR TITLE
[UI][Behat] tests for editing catalog promotion with rule and action

### PR DESCRIPTION
**Pull request by SirDomin**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets |
| License         | MIT

### CODE REVIEW

File: features/promotion/managing_catalog_promotions/editing_catalog_promotion.feature [hunk 1]
```diff
@@ -44,13 +44,13 @@ Feature: Editing catalog promotion
         When I want to modify a catalog promotion "Christmas sale"
         Then I should not be able to edit its code
 
-    @api
+    @api @ui @javascript
     Scenario: Editing catalog promotion rule
         When I want "Christmas sale" catalog promotion to be applied on "Kotlin T-shirt" variant
         Then I should be notified that it has been successfully edited
         And this catalog promotion should be applied on "Kotlin T-shirt" variant
```
Comment (GSadee) [inline features/promotion/managing_catalog_promotions/editing_catalog_promotion.feature:51]: What about a "PHP T-Shirt" variant? Is this promotion still applying to this variant? I would be for adding a step to check that

File: features/promotion/managing_catalog_promotions/editing_catalog_promotion.feature [hunk 2]
```diff
@@ -44,13 +44,13 @@ Feature: Editing catalog promotion
...
 
-    @api
+    @api @ui @javascript
     Scenario: Editing catalog promotion action
         When I edit "Christmas sale" catalog promotion to have "40%" discount
```
Comment (GSadee) [inline features/promotion/managing_catalog_promotions/editing_catalog_promotion.feature:55]: Would be nice to have a different discount already defined on this catalog promotion

File: features/promotion/managing_catalog_promotions/editing_catalog_promotion.feature [hunk 3]
```diff
@@ -44,13 +44,13 @@ Feature: Editing catalog promotion
         When I want to modify a catalog promotion "Christmas sale"
         Then I should not be able to edit its code
 
-    @api
+    @api @ui @javascript
     Scenario: Editing catalog promotion rule
         When I want "Christmas sale" catalog promotion to be applied on "Kotlin T-shirt" variant
```
Comment (GSadee) [inline features/promotion/managing_catalog_promotions/editing_catalog_promotion.feature:49]:
```suggestion
        When I edit "Christmas sale" catalog promotion to be applied on "Kotlin T-shirt" variant
```

File: src/Sylius/Behat/Context/Ui/Admin/ManagingCatalogPromotionsContext.php [hunk 4]
```diff
@@ -189,6 +191,30 @@ public function iSaveMyChanges(): void
...
+    /**
+     * @When /^I want ("[^"]+" catalog promotion) to be applied on ("[^"]+" variant)$/
+     */
+    public function iWantCatalogPromotionToBeAppliedOn(CatalogPromotionInterface $catalogPromotion, ProductVariantInterface $productVariant): void
+    {
+        $this->updatePage->open(['id' => $catalogPromotion->getId()]);
+        $variantCodes = [$productVariant->getCode()];
+
+        $this->formElement->addRule();
+        $this->formElement->chooseLastRuleVariants($variantCodes);
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingCatalogPromotionsContext.php:203]:
```suggestion
        $this->formElement->addRule();
        $this->formElement->chooseLastRuleVariants([$productVariant->getCode()]);
```

File: features/promotion/managing_catalog_promotions/editing_catalog_promotion.feature (hunk 2 above)
Comment (SirDomin) [reply features/promotion/managing_catalog_promotions/editing_catalog_promotion.feature:57]: Added another step to check that

**Review by GSadee — APPROVED** [review]

File: features/promotion/managing_catalog_promotions/editing_catalog_promotion.feature (hunk 1 above)
Comment (SirDomin) [reply features/promotion/managing_catalog_promotions/editing_catalog_promotion.feature:52]: Added another step to check that

**Review by arti0090 — APPROVED** [review]

**Review by lchrusciel — APPROVED** [review]

**Comment by lchrusciel** [comment]
Thanks, @SirDomin! :1st_place_medal:

**Pull request merged**